File: alms_app/crms/web_form/car_indent_form/car_indent_form.py
import frappe
from frappe.core.doctype.communication.email import make
from email.mime.text import MIMEText
import smtplib
from alms_app.api.emailsService import email_sender


@frappe.whitelist(allow_guest=True)
def get_employee_details(employee_code):
    employee = frappe.get_all("Employee Master", filters={"name": employee_code}, 
    fields=["name", "designation", "location", "company", "contact_number", "email_id", "department","eligibility","reporting_head"])
    
    employee_eligibility = frappe.get_doc("Employee Designation",employee[0].designation)
    employee.append({"eligibility":employee_eligibility.eligibility})
    if employee:
        employee_details = employee[0]
        return employee
    else:
        frappe.throw(f"Employee with name '{employee_code}' not found.")
        
@frappe.whitelist(allow_guest=True)
def calculate_totals(frm_data):
    ex_showroom_price = frm_data.get('ex_showroom_price', 0)
    discount = frm_data.get('discount', 0)
    tcs = frm_data.get('tcs', 0)
    registration_charges = frm_data.get('registration_charges', 0)
    accessories = frm_data.get('accessories', 0)
    
    net_ex_showroom_price = ex_showroom_price - discount + tcs
    finance_amount = net_ex_showroom_price + registration_charges + accessories
    
    result = {
        "net_ex_showroom_price": f"₹ {net_ex_showroom_price:,.2f}",
        "finance_amount": f"₹ {finance_amount:,.2f}"
    }
    
    return result

def get_context(context):
    context.employee_code = frappe.form_dict.get('employee_code')
    context.employee_details = {}

    if context.employee_code:
        employee_details = get_employee_details(context.employee_code)
        context.employee_details = employee_details

    return context

@frappe.whitelist(allow_guest=True)
def check_indent_exists(employee_code):
    """Check if a Car Indent Form already exists for the given employee."""
    print(f"Checking if Car Indent Form exists for employee code: {employee_code}")
    
    exists = frappe.db.exists("Car Indent Form", {"name": employee_code})
    
    print(f"Indent exists: {exists}")
    
    if exists:
        return "redirect"
    
    return False

@frappe.whitelist(allow_guest=True)
def send_email_to_reporting_head(doc, method=None):
    # print(doc,"000000000000000")
    try:
        employee_docname = doc
        # print(employee_docname,"000000000000000")
        # print(doc,"000000000000000")
        # print(doc,"000000000000000")

        email_sender(name=employee_docname, email_send_to="To Reporting", car_indent_form_name=doc)
        return {"status": "success", "message": f"Email triggered for reporting head of {employee_docname}"}
    except Exception as e:
        frappe.log_error(f"Failed to send reporting head email: {str(e)}", "Car Indent Email Error")
        return {"status": "error", "message": str(e)}

    except Exception as e:
        frappe.log_error(f"Error in send_email_to_reporting_head: {str(e)}")
        return {"status": "failed", "message": str(e)}

File: alms_app/crms/web_form/car_indent_form/car_indent_form.js
console.log("✅ car_indent_form.js loaded-----");

frappe.ready(function () {
    function getEmployeeCodeFromURL() {
        const params = new URLSearchParams(window.location.search);
        return decodeURIComponent(params.get("employee_code") || "");
    }

    const employeeCode = getEmployeeCodeFromURL();
    if (employeeCode) {
        frappe.web_form.set_value('employee_code', employeeCode);

        frappe.call({
            method: "alms_app.crms.web_form.car_indent_form.car_indent_form.get_employee_details",
            args: { employee_code: employeeCode },
            callback: function (response) {
                const employeeDetails = response.message;
                if (employeeDetails) {
                    frappe.web_form.set_value('designation', employeeDetails[0].designation);
                    frappe.web_form.set_value('eligibility', employeeDetails[0].eligibility);
                    frappe.web_form.set_value('ex_showroom_price', employeeDetails[0].eligibility);
                    frappe.web_form.set_value('location', employeeDetails[0].location);
                    frappe.web_form.set_value('company_name', employeeDetails[0].company);
                    frappe.web_form.set_value('employee_reporting', employeeDetails[0].reporting_head);
                    frappe.web_form.set_value('contact_number', employeeDetails[0].contact_number);
                    frappe.web_form.set_value('email_id', employeeDetails[0].email_id);
                    frappe.web_form.set_value('department', employeeDetails[0].department);
                }
            }
        });
    }

    frappe.web_form.on("ex_showroom_price", calculate_totals);
    frappe.web_form.on("discount", calculate_totals);
    frappe.web_form.on("tcs", calculate_totals);
    frappe.web_form.on("registration_charges", calculate_totals);
    frappe.web_form.on("accessories", calculate_totals);

    function formatINR(amount) {
        return new Intl.NumberFormat('en-IN', { style: 'currency', currency: 'INR' }).format(amount);
    }

    function calculate_totals() {
        const ex_showroom_price = parseFloat(frappe.web_form.get_value('ex_showroom_price')) || 0;
        const discount = parseFloat(frappe.web_form.get_value('discount')) || 0;
        const tcs = parseFloat(frappe.web_form.get_value('tcs')) || 0;
        const registration_charges = parseFloat(frappe.web_form.get_value('registration_charges')) || 0;
        const accessories = parseFloat(frappe.web_form.get_value('accessories')) || 0;

        const net_ex_showroom_price = ex_showroom_price - discount + tcs;
        const finance_amount = net_ex_showroom_price + registration_charges + accessories;

        frappe.web_form.set_value('net_ex_showroom_price', net_ex_showroom_price);
        frappe.web_form.set_value('finance_amount', finance_amount);
        frappe.web_form.set_value('net_ex_showroom_price_inr', formatINR(net_ex_showroom_price));
        frappe.web_form.set_value('finance_amount_inr', formatINR(finance_amount));
    }

    calculate_totals();

    frappe.web_form.validate = function () {
        const employeeCode = frappe.web_form.get_value("employee_code");
        return new Promise((resolve, reject) => {
            frappe.call({
                method: "alms_app.crms.web_form.car_indent_form.car_indent_form.check_indent_exists",
                args: { employee_code: employeeCode },
                callback: function (response) {
                    if (response.message === "redirect") {
                        frappe.msgprint("You already have an active indent.");
                        setTimeout(() => { window.location.href = "/already-present"; }, 2000);
                        reject("Indent already exists");
                    } else {
                        frappe.call({
                            method: "alms_app.crms.web_form.car_indent_form.car_indent_form.send_email_to_reporting_head",
                            args: { doc: employeeCode },
                            callback: function () { resolve(); },
                            error: function () {
                                frappe.throw("Error sending car indent email.");
                                reject("Email error");
                            }
                        });
                    }
                }
            });
        });
    };
});
